pinc/misc.inc

```diff
@@ -123,25 +123,30 @@ function get_changed_fields_for_objects($new, $old, $remove_non_public = true)
  * @param array $choices
  * @param bool $allownull
```

Comment:
PHPStan understands generics, so we could say that the return type has to be the same as the choices and the optional default with something like
```
@template T 
@param array $arr   # The array can be (and probably is) heterogeneous. Don't constraint the key/value types.
@param string $key
@param T|null $default
@param array<T> $choices
@param bool $allownull
@return T
```
In practice, I'm sure `T` will be `string` most of the time.

See https://phpstan.org/writing-php-code/phpdoc-types and https://phpstan.org/blog/generics-by-examples

Reply:
Nits: (I can't figure out how to add suggestions to lines that are outside the diff), but lines 105 & 106 talk about 'string'. This should be 'value'.

Reply:
Good catch, fixed in incoming commit.